vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -262,12 +263,12 @@ def _get_modified_page_figures(
         parameters=ctds_parameters,
     )
 
-    outputs = {}
+    outputs: Dict[ModelID, Any] = {}
     for target in targets:
         outputs[target] = model_manager[target](  # type: ignore[operator]
             data_frame=filtered_data[target], **parameterized_config[target]
         )
         if hasattr(outputs[target], "update_layout"):
             outputs[target].update_layout(template="vizro_dark" if ctd_theme["value"] else "vizro_light")
 
-    return outputs
+    return namedtuple("outputs", outputs.keys())(**outputs)
```

Comment:
I decided to handle the action outputs for all predefined actions using the `namedtuple` variant because then we don't need to care about the output order.